Library/Homebrew/formulary.rb

```diff
@@ -190,6 +191,38 @@ def self.from_rack(rack, spec=:stable)
     end
   end
 
+  # Return a Formula instance for oldname
+  def self.from_oldname(oldname)
```

Comment:
Referring to [my pervious comment](https://github.com/Homebrew/homebrew/pull/41006/files#r35743574), I think we don't need this method. It's the logic inside `Formulary#factory` you need to alter.